### PR TITLE
Updated GetPTTRequest to fully support and verify optional addresses

### PR DESCRIPTION
**Pull request by elichai**
- Updated the `core_messages_scheme.json` scheme to support and verify optional addresses in `GetPTTRequest`
- Added more tests for this part of the IPC
- Updated the existing principal tests and fixed some bugs.

This should Fix #123 and #157
@lacabra

### CODE REVIEW

File: src/core/core_server_mock/core_server.js [hunk 1]
```diff
@@ -50,11 +50,17 @@ class MockCoreServer {
 
   static _getPTTRequest(msg) {
     if (MockCoreServer._validate(msg, SCHEMES.GetPTTRequest)) {
+      let request;
+      if (msg.input && msg.input.addresses) {
+        request = msg.input.addresses;
+      } else {
+        request = 'no addresses';
```
Comment (lenak25) [inline src/core/core_server_mock/core_server.js]: Please set the `no address` string into a const

**Comment by elichai** [comment]
@lacabra If you change my code *please* ping me about it, comment here something, anything.

With the change you did we might as well skip this test. because if the `response` wasn't set correctly it will just skip this important verification.

**Comment by lacabra** [comment]
@elichai OK.

On another front, please avoid rewriting history, as it makes it harder on others to track the most recent changes, since we lose any references to previous contributions.

**Comment by elichai** [comment]
But otherwise the history Isn't ordered chronologically and has garbage in it.
If I try something and decide not to do it the right thing is to remove the commit, not create another one.

This way you avoid the whole "fix last commit" "fix again" "another fix" sharade.
And by rebasing onto develop instead of merging you make the commits be in the right order after it'll merge to develop (and you avoid another meaningless commit)

**Pull request merged**